pkg/client/connector/userd_k8s/k8s_cluster.go

```diff
@@ -101,37 +101,56 @@ func (kc *Cluster) check(c context.Context) error {
 	return c.Err()
 }
 
-// kindNames returns the names of all objects of a specified Kind in a given Namespace
-func (kc *Cluster) kindNames(c context.Context, kind, namespace string) ([]string, error) {
-	var objNames []objName
-	if err := kc.client.List(c, kates.Query{Kind: kind, Namespace: namespace}, &objNames); err != nil {
+// Deployments returns all deployments found in the given Namespace
+func (kc *Cluster) Deployments(c context.Context, namespace string) ([]kates.Object, error) {
+	var deployments []*kates.Deployment
+	if err := kc.client.List(c, kates.Query{Kind: "Deployment", Namespace: namespace}, &deployments); err != nil {
 		return nil, err
 	}
-	names := make([]string, len(objNames))
-	for i, n := range objNames {
-		names[i] = n.Name
+	objs := make([]kates.Object, len(deployments))
+	for i, dep := range deployments {
+		objs[i] = dep
 	}
-	return names, nil
+	return objs, nil
 }
 
-// DeploymentNames returns the names of all deployments found in the given Namespace
-func (kc *Cluster) DeploymentNames(c context.Context, namespace string) ([]string, error) {
-	return kc.kindNames(c, "Deployment", namespace)
-}
-
-// ReplicaSetNames returns the names of all replica sets found in the given Namespace
-func (kc *Cluster) ReplicaSetNames(c context.Context, namespace string) ([]string, error) {
-	return kc.kindNames(c, "ReplicaSet", namespace)
+// ReplicaSets returns all replica sets found in the given Namespace
+func (kc *Cluster) ReplicaSets(c context.Context, namespace string) ([]kates.Object, error) {
+	var replicaSets []*kates.ReplicaSet
+	if err := kc.client.List(c, kates.Query{Kind: "ReplicaSet", Namespace: namespace}, &replicaSets); err != nil {
+		return nil, err
+	}
+	objs := make([]kates.Object, len(replicaSets))
+	for i, rs := range replicaSets {
+		objs[i] = rs
+	}
+	return objs, nil
 }
 
-// StatefulSetNames returns the names of all replica sets found in the given Namespace
-func (kc *Cluster) StatefulSetNames(c context.Context, namespace string) ([]string, error) {
-	return kc.kindNames(c, "StatefulSet", namespace)
+// StatefulSets returns all stateful sets found in the given Namespace
+func (kc *Cluster) StatefulSets(c context.Context, namespace string) ([]kates.Object, error) {
+	var statefulSets []*kates.StatefulSet
+	if err := kc.client.List(c, kates.Query{Kind: "StatefulSet", Namespace: namespace}, &statefulSets); err != nil {
+		return nil, err
+	}
+	objs := make([]kates.Object, len(statefulSets))
+	for i, ss := range statefulSets {
+		objs[i] = ss
+	}
+	return objs, nil
 }
 
-// PodNames returns the names of all replica sets found in the given Namespace
+// PodNames returns the names of all pods found in the given Namespace
 func (kc *Cluster) PodNames(c context.Context, namespace string) ([]string, error) {
```

Comment:
I think we could make a `func (kc *Cluster) Pods` like you did for the other objects, since it's only called in `refreshReplicaSet`, we could delete this function.

And then refreshReplicaSet would be something like this (snippet has not been tested, it might work but think of it as just pseudocode to get my point across):
```
  353 func (ki *installer) refreshReplicaSet(c context.Context, namespace string, rs *kates.ReplicaSet) error {
  354     pods, err := ki.Pods(c, namespace)
  355     if err != nil {
  356         return err
  357     }
  358     for _, pod := range pods {
  359         for _, ownerRef := range pod.OwnerReferences {
  360             if ownerRef.UID == rs.UID {
  361                 dlog.Infof(c, "Deleting pod %s owned by rs %s", pod.Name, rs.Name)
  362                 pod := &kates.Pod{
  363                     TypeMeta: kates.TypeMeta{
  364                         Kind: "Pod",
  365                     },
  366                     ObjectMeta: kates.ObjectMeta{
  367                         Namespace: pod.Namespace,
  368                         Name:      pod.Name,
  369                     },
  370                 }
  371                 if err := ki.Client().Delete(c, pod, pod); err != nil {
  372                     return err
  373                 }
  374             }
  375         }
  376     }
  377     return nil
  378 }
  ```